src/partition/repart.c

```diff
@@ -1396,13 +1404,16 @@ static int context_read_definitions(
         _cleanup_strv_free_ char **files = NULL;
         Partition *last = NULL;
         int r;
+        const char *const *dirs;
 
         assert(context);
 
         if (directory)
-                r = conf_files_list_strv(&files, ".conf", NULL, CONF_FILES_REGULAR|CONF_FILES_FILTER_MASKED, (const char**) STRV_MAKE(directory));
+                dirs = STRV_MAKE_CONST(directory);
         else
-                r = conf_files_list_strv(&files, ".conf", root, CONF_FILES_REGULAR|CONF_FILES_FILTER_MASKED, (const char**) CONF_PATHS_STRV("repart.d"));
+                dirs = (const char* const*) CONF_PATHS_STRV("repart.d");
+
+        r = conf_files_list_strv(&files, ".conf", NULL, CONF_FILES_REGULAR|CONF_FILES_FILTER_MASKED, dirs);
```

Comment:
`root` is dropped mistakenly(?).

Reply:
Yes, it was a mistake. It should be fixed now.